utils/pip/utils.go

```diff
@@ -0,0 +1,181 @@
+// pythonExecPath - Execution path python.
+func runPythonCommand(execPath, command string, cmdArgs []string) (data []byte, err error) {
+	pipeReader, pipeWriter := io.Pipe()
+	defer func() {
+		e := pipeReader.Close()
+		if err == nil {
+			err = e
+		}
+	}()
+	log.Debug(fmt.Sprintf("Running python command: %s %s %v", execPath, command, cmdArgs))
+	// Execute the python pip-dependency-map script.
+	pipDependencyMapCmd := &piputils.PipCmd{
+		Executable:  execPath,
+		Command:     command,
+		CommandArgs: cmdArgs,
+		StrWriter:   pipeWriter,
+	}
+	var pythonErr error
+	go func() {
+		pythonErr = gofrogcmd.RunCmd(pipDependencyMapCmd)
+	}()
+	data, err = ioutil.ReadAll(pipeReader)
+	if err != nil {
+		return nil, errorutils.CheckError(err)
+	}
+	if pythonErr != nil {
+		return nil, errorutils.CheckError(pythonErr)
+	}
+	return data, err
+}
+
+// Execute virtualenv command. "virtualenv .jfrog"
+func RunVirtualEnv(venvDirPath string) (err error) {
+	execPath, err := GetExecutablePath("virtualenv")
+	_, err = runPythonCommand(execPath, venvDirPath, []string{})
+	return err
+}
+
+// Execute pip install command. "pip install ."
+func RunPipInstall(venvDirPath string) (err error) {
+	_, err = runPythonCommand(filepath.Join(venvDirPath, "bin", "pip"), "install", []string{"."})
+	return err
+}
+
+// Execute pip-dependency-map script, return dependency map of all installed pip packages in current environment.
+func RunPipDepTree(venvDirPath string) (map[string][]string, []string, error) {
+	pipDependencyMapScriptPath, err := GetDepTreeScriptPath()
+	if err != nil {
+		return nil, nil, err
+	}
+	data, err := runPythonCommand(filepath.Join(venvDirPath, "bin", "python"), pipDependencyMapScriptPath, []string{"--json"})
+
+	// Parse the result.
+	return parsePipDependencyMapOutput(data)
+}
+
+// Parse pip-dependency-map raw output to dependencies map.
+func parsePipDependencyMapOutput(data []byte) (map[string][]string, []string, error) {
+	// Parse into array.
+	packages := make([]pipDependencyPackage, 0)
+	if err := json.Unmarshal(data, &packages); err != nil {
+		return nil, nil, errorutils.CheckError(err)
+	}
+
+	// Create packages map.
+	packagesMap := map[string][]string{}
+	allSubPackages := map[string]bool{}
+	for _, pkg := range packages {
+		var subPackages []string
+		for _, subPkg := range pkg.Dependencies {
+			subPkgFullName := subPkg.Key + ":" + subPkg.InstalledVersion
+			subPackages = append(subPackages, subPkgFullName)
+			allSubPackages[subPkgFullName] = true
+		}
+		packagesMap[pkg.Package.Key+":"+pkg.Package.InstalledVersion] = subPackages
+	}
+
+	var topLevelPackagesList []string
+	for pkgName := range packagesMap {
+		if allSubPackages[pkgName] == false {
+			topLevelPackagesList = append(topLevelPackagesList, pkgName)
+		}
+	}
+	return packagesMap, topLevelPackagesList, nil
+}
+
+// Return path to the dependency-tree script, if not exists it creates the file.
+func GetDepTreeScriptPath() (string, error) {
```

Comment:
Looks like this function can be made private.
This function should also be changed, so that it creates the ```pipdeptree.py``` file if it doesn't exist already under the depedencies directory. The file should be created the first time it is needed, and resused.
Also - the file which holds the python script should have a version. so that we can force jfrog-cli to update the script content when the script code changes. This version should be name of the directory which stores the script under the dependencies directory.
We can start with version 1.
The same approach is taken when managing build-info-extractors of maven and gradle.